packages/thirdweb/src/wallets/wallet-connect/index.ts

```diff
@@ -258,18 +289,33 @@ async function initProvider(
 
   if (walletId !== "walletConnect") {
     function handleSessionRequest() {
-      if (typeof window === "undefined") {
+      const preferUniversal =
+        walletInfo.mobile.universal || walletInfo.mobile.native;
+      const preferNative =
+        walletInfo.mobile.native || walletInfo.mobile.universal;
+
+      if (isReactNative()) {
+        // TODO prefere universal on android?
+        const appUrl = walletInfo.mobile.native || walletInfo.mobile.universal;
+        if (appUrl && options.walletConnect?.onDisplayUri) {
+          options.walletConnect?.onDisplayUri(
+            formatWalletConnectUrl(appUrl, "").href,
+          );
+        } else {
+          throw new Error(
+            "No 'onDisplayUri' callback or app url found for wallet connect to redirect to.",
+          );
+        }
         return;
       }
 
-      if (!isMobile()) {
+      if (!isBrowser()) {
         return;
       }
 
-      const preferUniversal =
-        walletInfo.mobile.universal || walletInfo.mobile.native;
-      const preferNative =
-        walletInfo.mobile.native || walletInfo.mobile.universal;
+      if (!isMobile()) {
```

Comment:
If not browser (above if statement) and not mobile continue? What case would get through both of those

Reply:
fixed